x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java

```diff
@@ -93,6 +93,14 @@ public void testCIDRMatchNonString() {
                 "line 1:15: argument of [cidrMatch(source_address, 12345)] must be [string], found value [12345] type [integer]", msg);
     }
 
+    public void testConcatWithInexact() {
+        VerificationException e = expectThrows(VerificationException.class,
+            () -> plan("process where concat(plain_text)"));
+        String msg = e.getMessage();
+        assertEquals("Found 1 problem\nline 1:15: [concat(plain_text)] cannot operate on first argument field of data type "
```

Comment:
Small improvement here: if it's possible for a one argument call to not use "first" in the error message, that would look more user friendly.

Also, I would like to see one or two tests where another argument different from the first doesn't fulfill the restriction.